### DISCUSSION TITLE
Access the config in multiple python files

### DISCUSSION THREAD

**Discussion by charbel-el-khoury** [facebookresearch/hydra#2858] [Q&A]
Is it possible to access the config in multiple modules of a large project.

For example, if I have the following file structure:
```
project_root/
├── conf/
│   └── config.yaml
└── src/
│   ├── module1.py
│   ├── module2.py
└──main.py
```
What would be the best practice to access config variables in the functions and classes of `module1.py` and `module2.py`? Right now I am passing the `DictConfig` object as a parameter of the classes and functions in the module. Is there a bettter way to do this?

**Comment by odelalleau** [comment]
> Right now I am passing the DictConfig object as a parameter of the classes and functions in the module.

That's the most common pattern AFAIK. But nothing prevents you from implementing a `cfg()` utility function that would return the config, or having some kind of "global state" object that would hold the config as `global_state.cfg`. This is all stuff you need to implement on your own though, I don't think Hydra provides something out of the box for this (though I may be wrong).

**Reply by Jasha10** [reply]
That's correct, Hydra does not provide dependency injection. The primary APIs for accessing your config are `@hydra.main` and `hydra.compose`. Once you get your config from one of those APIs, it's your responsibility to pass the config to the different parts of your app.
